syntaxes/graphql.js.json

```diff
@@ -3,43 +3,40 @@
   "injectionSelector": "L:source -string -comment",
   "patterns": [
     {
-      "name": "taggedTemplates",
       "contentName": "meta.embedded.block.graphql",
-      "begin": "(Relay\\.QL|gql|graphql(\\.experimental)?)\\(?(`)",
+      "begin": "(?:Relay\\.QL|gql|graphql(\\.experimental)?)\\s*(?:<.*?>\\s*)?`",
+      "end": "`",
+      "patterns": [{ "include": "source.graphql" }]
+    },
+    {
+      "contentName": "meta.embedded.block.graphql",
+      "begin": "`(\\s*#[ ]*(gql|graphql|GraphQL))",
       "beginCaptures": {
         "1": {
-          "name": "entity.name.function.tagged-template.js"
+          "name": "meta.embedded.block.graphql comment.line.graphql.js"
         },
         "2": {
-          "name": "punctuation.definition.string.template.begin.js"
```

Comment:
I did remove these `entity.name.function.tagged-template.js` and `punctuation.definition.string.template.begin.js` since we don't want *our* grammar to attack tokens and highlight those, but we want the parent language to add them.
Instead, for only this pattern, we add `meta.embedded.block.graphql comment.line.graphql.js"`, as the `source.graphql` matcher will only start working between start and end pattern, but we also want the GraphQL in the start pattern to be properly highlighted.

The other patterns don't add capture names, apart from `markup.italic` on the word `gql/graphql/GraphQL` inside of comments.